File: Interpreted/mutual-recursion/mrec.pl

use experimental qw(signatures);

sub F($n) { $n == 0 ? 1 : $n - M(F($n-1)) }
sub M($n) { $n == 0 ? 0 : $n - F(M($n-1)) }

print F($ARGV[0]), "\n";
print M($ARGV[0]), "\n";

File: Interpreted/mutual-recursion/mrec.php
<?php
function F($n) {
  if ($n == 0) return 1;
  return $n - M(F($n-1));
}

function M($n) {
  if ($n == 0) return 0;
  return $n - F(M($n-1));
}

echo F($argv[1]), "\n";
echo M($argv[1]), "\n";
?>
